subprojects/core/src/main/java/org/gradle/api/file/FileTreeElement.java

```diff
@@ -103,4 +103,9 @@
      * @return The Unix file permissions.
      */
     int getMode();
+
+    /**
+     * @return true if the file is a regular file or directory on disk. False for files in zip or tar files.
+     */
+    boolean isPhysicalFile();
```

Comment:
So you are saying we should close the tarTree/zipTree somewhere else so that while we do the copying we have access to the input streams? Do you have an idea where to do that?